File: flutter/platform/lib/src/modules/ai.dart
import '../core/http_client.dart';
import '../core/api_endpoints.dart';
import '../core/result.dart';

/// AI module — text completion, model listing, usage, balance.
class AiModule {
  const AiModule();

  Future<ApiResult<Map<String, dynamic>>> complete(Map<String, dynamic> body) =>
      HttpClient.instance.postMap(ApiEndpoints.aiTextCompletion, body: body);

  Future<ApiResult<Map<String, dynamic>>> listModels() =>
      HttpClient.instance.getMap(ApiEndpoints.aiModels);

  Future<ApiResult<Map<String, dynamic>>> getUsage({
    String? period,
    String? from,
    String? to,
  }) =>
      HttpClient.instance.getMap(
        ApiEndpoints.aiUsage,
        queryParams: {
          if (period != null) 'period': period,
          if (from != null) 'from': from,
          if (to != null) 'to': to,
        },
      );

  Future<ApiResult<Map<String, dynamic>>> getBalance() =>
      HttpClient.instance.getMap(ApiEndpoints.aiBalance);
}

File: web/platform/src/modules/ai.ts
import { http } from "../core/http";
import { Endpoints } from "../core/endpoints";
import type { ApiResult } from "../core/result";

export const ai = {
  complete(body: {
    model: string;
    messages: Array<{ role: string; content: string; toolCallId?: string; name?: string }>;
    temperature?: number;
    topP?: number;
    maxTokens?: number;
    stop?: string[];
    jsonMode?: boolean;
    seed?: number;
    stream?: boolean;
    reasoningEnabled?: boolean;
    reasoningBudgetTokens?: number;
    webSearchEnabled?: boolean;
  }): Promise<ApiResult<Record<string, unknown>>> {
    return http.postMap(Endpoints.aiTextCompletion, body);
  },

  /** Raw streaming completion. Returns the Response for SSE consumption. */
  streamCompletion(body: {
    model: string;
    messages: Array<{ role: string; content: string }>;
    temperature?: number;
    maxTokens?: number;
  }): Promise<Response | null> {
    return http.postRaw(Endpoints.aiTextCompletion, { ...body, stream: true });
  },

  listModels(): Promise<ApiResult<Record<string, unknown>>> {
    return http.getMap(Endpoints.aiModels);
  },

  getUsage(query?: { period?: string; from?: string; to?: string }): Promise<ApiResult<Record<string, unknown>>> {
    return http.getMap(Endpoints.aiUsage, query);
  },

  getBalance(): Promise<ApiResult<Record<string, unknown>>> {
    return http.getMap(Endpoints.aiBalance);
  },
};
